### ISSUE TITLE
Fix false positives for inlines styles in custom-property-empty-line-before

### ISSUE THREAD

**Issue by dpolian** [stylelint/stylelint#8029]
### What minimal example or steps are needed to reproduce the bug?

<span  style="--grid-rows: 39;">
  Text
</span>

### What minimal configuration is needed to reproduce the bug?

{
  "customSyntax": "postcss-html",
  "rules": {
		"custom-property-empty-line-before": [
			"always",
			{
				"ignore": [
					"inside-single-line-block",
					"first-nested"
				]
			}
		]
  }
}

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnADgQwHYAIUBcBPAGxgF4AdEAWmoHMAnASymoYgHcV4cBmATgDcVAHwVcOACowAHvnFIA9OmwiQAGhAAzJqQByGALZxEso2lIA6ABb5DxDeAhYddBCGDicOKmACuBBCGAMqEWPgYMlQ8VGgQBGAoKNS29lTqXj4gDH6kKNE4nlgU8vLgAfhB1GjsaDAMRNQwhmiNxExYMNQARjBaEAwwBQDa4iXjZRjEHBiE+RpjpSVFE6sgTHRYA0MIOKPFS4dUHSgsXadYdKTU7Z09xBBgANbpi6tL2kwMBNSdBDBQKhvQ4AXWBJQAvsCwRIoVgIY5IC4NgAxAaGDD4dwAKxQzkcsDQKHcRW8sXi+ESyVSxAKVGImJgBFeEioBBIMFuZRiIAZ+CZZQyrJA7NIXOoSNc1AI2CgGAYgN29MZzJA4gREKAA)

### Which Stylelint-related dependencies are you using?

{
  "postcss-html": "latest",
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}

Installed:
[postcss-html](https://github.com/ota-meshi/postcss-html#readme)@1.7.0
[stylelint](https://stylelint.io/)@16.9.0
[stylelint-config-standard](https://www.npmjs.com/package/stylelint-config-standard)@36.0.1

### What did you expect to happen?

No errors to be detected.

### What actually happened?

```
1:14-30 error Expected empty line before custom property (custom-property-empty-line-before)
```

### Do you have a proposal to fix the bug?

looks similar to [#4658](https://github.com/stylelint/stylelint/issues/4658)

**Comment by ybiquitous** [comment]
@dpolian Thanks for the report with a reproducible demo! We can also reproduce it with only standard syntax, see [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIC0SBmELwAQCYBOMAtgNwA6AdiADQioCWANjAHICGhciMAHuwA5MAdGADOI6uAjl6AcwQhgFLFlIg8AVyYjV2ReWXLVYdSIAuEQkj54IfGHlMBPJET5OkDOuRhIARjHQCHSwAbSUDQxBWBgB3VkdtanCIvQiI1ToZcggghFDktMj6PDMkbzMYKFUqAsKM8hE6WCRG8hkmDy8fXwYIMABrVVqDAF1hgF8Csf0sSfJxiUhpTIAxHMJWU3kAKxEpCVg+cURUlRAzRyZPclNg1QZNmDMh+ZBxoA):

```css
--foo: 2rem;
```

I think the cause is that the [`ignore: ["inside-single-line-block"]` secondary option](https://github.com/stylelint/stylelint/tree/ab2141ef38e97e47c27202493512fb0c6ff550a7/lib/rules/custom-property-empty-line-before#inside-single-line-block) doesn't support a root node as a parent. See the relevant code:

https://github.com/stylelint/stylelint/blob/ab2141ef38e97e47c27202493512fb0c6ff550a7/lib/rules/custom-property-empty-line-before/index.mjs#L74-L82

The following change would suppress the bug, but I doubt `inside-single-line-block` may not fit the option name `inside-single-line-block` because a root node is not a block (which starts with `{` according to the [CSS spec](https://drafts.csswg.org/css-syntax/#consume-block)). 🤔 

```diff
- (isAtRule(parent) || isRule(parent)) &&
+ (isAtRule(parent) || isRule(parent) || isRoot(parent)) &&
```

Instead, it seems better to have a new option like `inside-single-line`. What do you think?

**Comment by ybiquitous** [comment]
FYI, PostCSS parses the code `--foo: 1px;` like this:

```sh-session
$ node -pe 'require("postcss").parse("--foo: 1px;").nodes[0]'
<ref *1> Declaration {
  raws: { before: '', between: ': ' },
  type: 'decl',
  parent: Root {
    raws: { semicolon: true, after: '' },
    type: 'root',
    nodes: [ [Circular *1] ],
    source: { input: [Input], start: [Object], end: [Object] },
    [Symbol(isClean)]: false,
    [Symbol(my)]: true
  },
  source: {
    input: Input {
      css: '--foo: 1px;',
      hasBOM: false,
      id: '<input css aaEBT8>',
      [Symbol(fromOffsetCache)]: [Array]
    },
    start: { column: 1, line: 1, offset: 0 },
    end: { column: 11, line: 1, offset: 11 }
  },
  prop: '--foo',
  value: '1px',
  [Symbol(isClean)]: false,
  [Symbol(my)]: true
}
```

**Comment by ybiquitous** [comment]
Good point. So, how about adding `ignore: ["first-line"]`? E.g.

```json
{
  "custom-property-empty-line-before": ["always", {"ignore": ["first-line"]}]
}
```

```css
--foo: 1px; /* ✅ not reported */
--bar: 2px; /* ❌ reported */
```

**Comment by jeddy3** [comment]
@dpolian Thanks for the report.
@ybiquitous Thanks for looking into it.

> Do you have a proposal to fix the bug?
> looks similar to https://github.com/stylelint/stylelint/issues/4658

Rather than introduce a new option specific to HTML syntax, shall we align the behaviour of `custom-property-empty-line-before` with `declaration-empty-line-before`? 

https://github.com/stylelint/stylelint/blob/74f3c080f20302c724659aaac24d47fcc713ab49/lib/rules/declaration-empty-line-before/index.mjs#L66-L69

As declarations and custom properties can't be the first node of root in CSS files.
